### PR TITLE
DeploymentManager - SDK preview

### PR DESCRIPTION
**Pull request by deveshguha**
Initial version of Deployment Manager SDK.

### CODE REVIEW

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (dsgouda) [inline src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Please rename this file as `DeploymentManager.Tests`

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (dsgouda) [inline src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Please set this as `1.0.0`

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (dsgouda) [inline src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Why is this dependency required. Also, reference a nuget package instead of a different project like so.

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (dsgouda) [inline src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Some comment as before for `Management.Authorization`

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (dsgouda) [inline src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Just set this to all json files under Tests\

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please regenerate the code using generate.ps1 and commit changes.

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (deveshguha) [reply src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Keeping this the same like we discussed as we're in preview.

Comment (deveshguha) [reply src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Setting this to 1.0.0 as this is a test project.

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (deveshguha) [reply src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: This dependency is required to have scenario tests for certain resources. Same for all such dependencies introduced.

Comment (dsgouda) [reply src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: If and when these RPs release their next versions they might break you. Please reference their published Nuget packages in order to avoid it.

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj
Comment (deveshguha) [reply src/SDKs/DeploymentManager/DeploymentManager.Tests/Management.DeploymentManager.Tests.csproj]: Fixed it.

**Comment by deveshguha** [comment]
Incorporated feedback.

**Comment by deveshguha** [comment]
Autogenerated using generate.ps1. 
This required a change to readme.txt in rest-api-specs repo to have the right output directory. This change is in the PR https://github.com/Azure/azure-rest-api-specs/pull/5489

File: src/SDKs/DeploymentManager/DeploymentManager.Tests/DeploymentManager.Tests.csproj
Comment (dsgouda) [inline src/SDKs/DeploymentManager/DeploymentManager.Tests/DeploymentManager.Tests.csproj]: Remove `Project` and `Name` tags

File: src/SDKs/DeploymentManager/AzSdk.RP.props
Comment (dsgouda) [inline src/SDKs/DeploymentManager/AzSdk.RP.props]: Please run `msbuild build.proj /t:build /p:Scope=SDKs\DeploymentManager` and commit all changes generated

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great, a couple of minor changes and should be good to go

**Review by dsgouda — APPROVED** [review]
LGTM will merge once CIs pass

**Review by dsgouda — CHANGES_REQUESTED** [review]
Spec is not merged. Approved in principle. @deveshguha Please regenerate the code once the spec is merged. Also, the .txt file generated is incorrect. Please make sure you delete and commit the one that is generated when you run the generate.ps1 script

**Comment by deveshguha** [comment]
> Spec is not merged. Approved in principle. @deveshguha Please regenerate the code once the spec is merged. Also, the .txt file generated is incorrect. Please make sure you delete and commit the one that is generated when you run the generate.ps1 script

Regenerated after the spec was merged.

**Review by dsgouda — APPROVED** [review]
LGTM will merge once CIs pass

**Comment by deveshguha** [comment]
> LGTM will merge once CIs pass

Sounds good, thank you!

**Pull request merged**
